compile error of arcus-3.5.8 branch

centos7, gcc 4.8.5 환경에서 arcus-3.5.8 클라이언트 코드를 컴파일 할 경우 다음과 같은 컴파일 에러가 발생한다.

```
src/zookeeper.c: In function 'zookeeper_change_ensemble':
src/zookeeper.c:5264:13: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
             char *temp = zh->hostname;
             ^
```

Makefile.am 파일에서 -Wdeclaration-after-statement 플래그를 삭제하면 정상적으로 컴파일 된다.

macOS, LLVM 10.0.1 환경에서는 플래그를 삭제하지 않아도 컴파일 되는 것으로 보아 gcc 버전에 관련된 이슈로 보인다.

--- a/zookeeper-client/zookeeper-client-c/src/zookeeper.c
+++ b/zookeeper-client/zookeeper-client-c/src/zookeeper.c
@@ -5260,8 +5260,9 @@ int zookeeper_change_ensemble(zhandle_t *zh, const char *hostname)
     if (rc == ZOK) {
         rc = zoo_set_servers(zh, hostname);
         if (rc != ZOK) {
+            char *temp;
             lock_reconfig(zh);
-            char *temp = zh->hostname;
+            temp = zh->hostname;
             zh->hostname = old_hostname;
             old_hostname = temp;
             unlock_reconfig(zh);

